query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/converters/PinotFunctionConverter.java

```diff
@@ -42,6 +43,8 @@ public String convert(
         return this.functionToString(this.toPinotPercentile(function), argumentConverter);
       case QUERY_FUNCTION_CONCAT:
         return this.functionToString(this.toPinotConcat(function), argumentConverter);
+      case QUERY_FUNCTION_AVG_RATE:
+        return this.functionToString(getAvgRateFunction(function), argumentConverter);
```

Comment:
Is it possible to contain all the logic in this class? That's what it's here for, to separate the function abstraction out of the main translation logic. For example, could we sum and divide all in pinot (I would guess yes, but not sure)? If not, what can we do to limit the changes in the handler and sql converter classes?

Reply:
We need to operate on both request and response which are present in handler only.

Reply:
> sum and divide not available).

So this is the assumption I'm getting at, because doing these in pinot makes things _far_ cleaner and as far as I can tell, they are available - as long as sum is outer because we need to aggregatewhen grouping. Luckily for us, sum of a divided value is the same as a divided sum.

For example:

```sql
Select 
    SUM(DIV(duration_millis, 60)),
	SUM(duration_millis),
    dateTimeConvert(start_time_millis,'1:MILLISECONDS:EPOCH','1:MILLISECONDS:EPOCH','60:SECONDS')
FROM spanEventView 
WHERE customer_id = 'X' 
	AND start_time_millis > 1632492379724 and start_time_millis < 1632511502000
GROUP BY  
    dateTimeConvert(start_time_millis,'1:MILLISECONDS:EPOCH','1:MILLISECONDS:EPOCH','60:SECONDS') 
limit 100
```

![image](https://user-images.githubusercontent.com/45047841/134730321-db5d9fb6-3aab-49ec-aee0-025c2205c356.png)

Reply:
This looks great and exactly what we want. I was not aware of this feature in pinot. Let me try this out.

Reply:
Thanks @aaron-steinfeld for providing with the feedback !!
SUM(DIV(..)) reduces a lot of complexity now.